src/ToolBox/Metadata.php

```diff
@@ -11,13 +11,15 @@
 {
     /**
      * @param ParameterDefinition|null $parameters
+     * @param array<mixed>|null        $responseFormat
```

Comment:
The resolving of the responseFormat should be done before, so that we just have to deal with null (no response format) or a valid response format (array) here in the `Metadata` class, I hope you agree on that